code/__HELPERS/lists.dm

```diff
@@ -855,3 +855,23 @@
 			return FALSE
 
 	return TRUE
+
+// Pick something else from a list than we last picked
+/proc/pick_different_from_list(list/l, current_index)
+	var/next_index
```

Comment:
I think this could just be simplified down to pick(list - list[index]), as list - list[index] should result in a list without the item

Reply:
Not sure whichever is faster, I was worried that if the list is huge, then list operations might take a lot of time.

If we could get any further input on this, it'd be appreciated

Reply:
it will be much faster just to subtract the prev color from the list

Reply:
Done in 9716394